src/Raven.Client/Document/Subscription.cs

```diff
@@ -32,289 +41,444 @@ namespace Raven.Client.Document
+        public event BeforeBatch BeforeBatch = delegate { };
+        public event AfterBatch AfterBatch = delegate { };
+        public event BeforeAcknowledgment BeforeAcknowledgment = () => true;
+        public event AfterAcknowledgment AfterAcknowledgment = delegate { };
+
         private void Start()
         {
-            StartWatchingDocs();
             startPullingTask = StartPullingDocs();
         }
 
+        private class WebSocketReadStream : Stream
+        {
+            private readonly RavenClientWebSocket _webSocket;
+            private readonly CancellationToken _token;
+
+            public WebSocketReceiveResult LastResult;
+
+            public WebSocketReadStream(RavenClientWebSocket webSocket, CancellationToken token)
+            {
+                _webSocket = webSocket;
+                _token = token;
+            }
+
+            public override void Flush()
+            {
+                throw new NotImplementedException();
+            }
+
+            public override int Read(byte[] buffer, int offset, int count)
+            {
+                throw new NotImplementedException();
+            }
+
+            public override long Seek(long offset, SeekOrigin origin)
+            {
+                throw new NotImplementedException();
+            }
+
+            public override void SetLength(long value)
+            {
+                throw new NotImplementedException();
+            }
+
+            public override void Write(byte[] buffer, int offset, int count)
+            {
+                throw new NotImplementedException();
+            }
+
+            public override async Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+            {
+                var receiveAsync = await _webSocket.ReceiveAsync(new ArraySegment<byte>(buffer, offset, count), _token).ConfigureAwait(false);
+                LastResult = receiveAsync;
+                return receiveAsync.Count;
+            }
+
+
+
+            public override bool CanRead => true;
+            public override bool CanSeek { get; }
+            public override bool CanWrite { get; }
+            public override long Length { get; }
+            public override long Position { get; set; }
+        }
+
+
+
+
+
         private Task PullDocuments()
         {
             return Task.Run(async () =>
             {
+                var queue = new BlockingCollection<RavenJObject>();
                 try
                 {
-                    long? lastProcessedEtagOnClient = null;
+                    var uri = new Uri(CreatePullingRequest().Url.Replace("http://", "ws://").Replace(".fiddler", ""));
 
-                    while (true)
+                    using (var ms = new MemoryStream())
                     {
-                        anySubscriber.WaitOne();
+                        ms.SetLength(1024 * 4);
+                        await webSocket.ConnectAsync(uri, cts.Token).ConfigureAwait(false);
```

Comment:
You are actually doing a LOT more than just doing ConfigureAwait in this commit, the commit message is misleading